### ISSUE TITLE
[Suspend] Popup doesn't work, gives g.username is not a function

### ISSUE THREAD

**Issue by Luceos** [flarum/core#682]
An issue that has plagued the live forum (tags is running dev-master) is that we are unable to suspend people. The source gives an error on 

> Uncaught TypeError: g.username is not a function

This reference is found twice in the forum-n.js:

```
System.register("flarum/helpers/avatar",[],function(h){h("default",function(g){var f=1>=arguments.length||void 0===arguments[1]?{}:arguments[1];f.className="Avatar "+(f.className||"");var e="",a="undefined"===f.title||f.title;a||delete f.title;if(g){var e=g.username()||"?",b=g.avatarUrl();a&&(f.title=f.title||e);if(b)return m("img",babelHelpers._extends({},f,{src:b}));e=e.charAt(0).toUpperCase();f.style={background:g.color()}}return m("span",f,e)});return{setters:[],execute:function(){}}});
System.register("flarum/helpers/username",[],function(h){h("default",function(g){g=g&&g.username()||app.translator.trans("core.lib.username.deleted_text");return m("span",{className:"username"},g)});return{setters:[],execute:function(){}}});
```

The solutions I now use:

- delete the user in case of bot
- edit password; but that doesn't help anything because the session lives on

**Comment by Luceos** [comment]
The only element since the previous release of Suspend with username in it, is:

https://github.com/flarum/suspend/compare/v0.1.0-beta.3...master#diff-20d13a11d8b92aea4af118d8e9bd720aR27

**Comment by tobscure** [comment]
Fixed: c81a34a2c7939b3ce653f6b6e7fcc812f4442508

@dcsjapan and everyone – FYI (didn't expect you to know this... it's something I forgot to tell you!) – as a matter of future-proofing, I've made it so that if there's a variable called `user` passed into a translation, the translator will expect it to be a full `User` object (with all of a user's information) rather than just a username string. Then it will render the username into a new variable called `username`. So to illustrate:

```js
const user = app.store.getById('users', 1);

app.translator.trans('foo', {user: user});
```

```yml
foo: Hello {username}
```

**Comment by dcsjapan** [comment]
Ahh, very interesting!
